lib/_stream_readable.js

```diff
@@ -134,6 +134,8 @@ function ReadableState(options, stream, isDuplex) {
 
   // The number of writers that are awaiting a drain event in .pipe()s
   this.awaitDrain = 0;
+  // Use a WeakSet to ref the piped dest which has already been counted on `this.awaitDrain`
+  this.awaitDrainDestSet = new WeakSet();
```

Comment:
Why is this a `WeakSet` and not a `Set`? The pipe should keep the destination alive from the source anyway, to the two should be functionally equivalent, with `Set` having the added niceness of having a deterministic size that could replace `awaitDrain`.